tasks/src/app/simple-button/simple-button.component.ts

```diff
@@ -1,22 +1,40 @@
-import { Component, OnInit } from '@angular/core';
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
 
 @Component({
 	selector: 'simple-button',
 	templateUrl: './simple-button.component.html',
-	styleUrls: ['./simple-button.component.css']
+	styleUrls: ['./simple-button.component.css'],
+	changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class SimpleButtonComponent implements OnInit {
-	public numberOfClicks: number;
+	public inputStatus: boolean;
+	public showedText: string;
+	public saveShowedText: string;
 
 	public constructor() {
-		this.numberOfClicks = 0;
+		this.inputStatus = false;
+		this.saveShowedText = "";
+		this.showedText = "";
 	}
 
 	public ngOnInit(): void {
 	}
 
-	public increseNumberOfCounts(): void {
-		this.numberOfClicks++;
+	public changeInputStatus(): void {
+		this.inputStatus = !this.inputStatus;
+		if (this.inputStatus === false) {
+			this.saveShowedText = this.showedText;
+			this.showedText = "";
+		} else this.showedText = this.saveShowedText;
+
+	}
+
+	public handleKeyUp(event: any): void {
+		if (this.inputStatus) {
+			this.showedText = "greetings " + event.target.value;
+		} else {
+			this.showedText = "";
+		}
```

Comment:
```suggestion
		this.greetingText = event.target.value;
```

it is enough for this task